### ISSUE TITLE
LightGBM 2.3.1: LGBMRegressor has no argument init_model

### ISSUE THREAD

**Issue by sachin-smart** [microsoft/LightGBM#3057]
Operating System: MacOS Catalina 10.15.4
Python version: 3.8.2
LightGBM version: 2.3.1
Python package

While it is difficult to create a reproducible example, I can explain my steps. I am rather novice, so please let me know if there is something wrong with my process. I would like to include a LightGBM installation as part of a conda environment and use LGBMRegressor from the sklearn wrapper, so I tried the following:

1. `conda install lightgbm`, which installs version 2.3.1 and runs properly except that for `LGBMModel` (and also `LGBMRegressor`) do not have init_model arguments unlike what the source code shows on github. See below for an example of the differences across the two.

2. After `conda install lightgbm` did not work, I uninstalled it and deleted any files I could see related to it, and tried installing using Homebrew (`brew install lightgbm`) as per the LightGBM Installation Guide. Although this does not seem to allow me to install lightgbm as part of my environment, I wanted to see if init_model existed here anyways, given it is the recommended installation method. However, I encountered the same issue of no init_model arguments.

3. I again uninstalled everything from Homebrew (and deleted any related files), and tried `pip install lightgbm`. Again the same issues.

Looking in the code in the environment files when I had installed with conda or pip, I could see that these arguments didn't exist in the `fit()` methods of the `LGBMModel` or `LGBMRegressor` classes. However, they did exist in `train()` underneath the wrapper. Also, they existed in the code when I cloned the git folder manually and checked the sklearn.py file there. Thus, I manually added the init_model argument to sklearn.py from the conda installation in the conda environment. This worked perfectly, and I was able to train models, stop, then continue training them as I would expect. However, this is a rather hacky fix and only works if I can access the sklearn.py file, which is not possible for one of my uses (running on a cluster).

Is there any reason why the source code on Github for what looks like version 2.3.1 would be different from the code that is installed? And is it possible to add these arguments and update the installation version so that I can just update the lightgbm version (presumably to 2.3.2) and use that? That may be an silly question so excuse any ignorance!

[LGBMRegressor.fit() method](https://github.com/microsoft/LightGBM/blob/18c706dc840c72341a2a85e05dfb5fad6366902a/python-package/lightgbm/sklearn.py#L380) from source code on github:
```python 
   def fit(self, X, y,
            sample_weight=None, init_score=None, group=None,
            eval_set=None, eval_names=None, eval_sample_weight=None,
            eval_class_weight=None, eval_init_score=None, eval_group=None,
            eval_metric=None, early_stopping_rounds=None, verbose=True,
            feature_name='auto', categorical_feature='auto',
            callbacks=None, init_model=None):
```

LGBMRegressor.fit() method from sklearn.py in installation:
```python
    def fit(self, X, y,
            sample_weight=None, init_score=None, group=None,
            eval_set=None, eval_names=None, eval_sample_weight=None,
            eval_class_weight=None, eval_init_score=None, eval_group=None,
            eval_metric=None, early_stopping_rounds=None, verbose=True,
            feature_name='auto', categorical_feature='auto', callbacks=None):
```

Please note the missing `init_model=None` argument.

Let me know if there is anything I can clarify or add. Thanks so much for the package anyways! It's really very cool.

**Comment by StrikerRUS** [comment]
@sachin-smart Thanks for using LightGBM in your project!

1. > `conda install lightgbm`
   
   Unfortunately, we do not maintain conda recipe.

2. > `brew install lightgbm`

   Hombrew installs only CLI version, so it is not currently possible to use scikit-learn interface with it.

3. > `pip install lightgbm`

   It is the right way to install LightGBM with scikit-learn wrapper! `init_model` argument was added to the sklearn wrapper in #2447 (merged into `master` in f3afe98bedd469b04758024c93c9ad77a43dc195 5 Dec 2019) but the [`2.3.1` release](https://github.com/microsoft/LightGBM/releases/tag/v2.3.1) was made 28 Nov 2019. That is why you cannot find that argument in your codebase. You can try to install LightGBM from the latest sources: https://github.com/microsoft/LightGBM/tree/master/python-package#install-from-github or wait for a next release.

**Comment by sachin-smart** [comment]
Thank you for the clarification!
